model/model_training/custom_datasets/dialogue_collator.py

```diff
@@ -29,18 +31,26 @@ def __call__(self, features):
         for messages in features:
             messages = list(messages)
 
-            # Add a way for the model to terminate generation
-            # When we predict the start of a new expected question, we want to be able to stop generation
-            messages.append(self.tokenizer.eos_token)
+            if random.random() < self.random_offset_probability:
+                truncation = TruncationStrategy.DO_NOT_TRUNCATE
+                max_length = None
+            else:
+                truncation = TruncationStrategy.LONGEST_FIRST
+                max_length = self.max_length
+
+            # append eos token to each messages
+            assert self.tokenizer.eos_token
+            messages = [m + self.tokenizer.eos_token for m in messages]
```

Comment:
Did we converge to this? Using the end of prompt-reply the same as eos? Not totally against it if eveyone agrees

Reply:
EOS at end of messages simplifies the decoding, TRLX also relies on eos tokens (i.e. would either have to be patched or eos could be set to `<human>`). The format is a compromise between framing format v3 and  old `<bot> <user>` format v2.